crates/rpc-types-eth/src/transaction/request.rs

```diff
@@ -390,8 +434,10 @@ impl TransactionRequest {
             missing.push("to");
         }
 
-        if self.sidecar.is_none() {
+        // Either `sidecar` or `blob_versioned_hashes` must be set.
+        if self.sidecar.is_none() && self.blob_versioned_hashes.is_none() {
             missing.push("sidecar");
+            missing.push("blob_versioned_hashes");
```

Comment:
this is a bit of a footgun, because this now also allows without sidecar which was not the original intention of this because this is supposed to be a transactionbuilder and 4844 without sidecar can't be sent.

I'd like to keep it that way but rather expose another transactionbuilder function explicitly for this usecase

Reply:
I've reverted changes to the legacy, now it should be the same as before (instead of the panics).
I've left a TODO/question for you in the code, please let me know what you think.

Introduced a dedicated `TryFrom` to handle the conversion I need.